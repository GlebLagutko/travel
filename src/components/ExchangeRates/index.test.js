import {render} from '@testing-library/react';

import ExchangeWidget from '.';
import React from "react";
import {rest} from 'msw'
import {setupServer} from 'msw/node'

describe('ExchangeWidget', () => {

  const response ={
    result: "success",
    documentation: "https://www.exchangerate-api.com/docs",
    terms_of_use: "https://www.exchangerate-api.com/terms",
    time_last_update_unix: 1615852801,
    time_last_update_utc: "Tue, 16 Mar 2021 00:00:01 +0000",
    time_next_update_unix: 1615939216,
    time_next_update_utc: "Wed, 17 Mar 2021 00:00:16 +0000",
    base_code: "NOK",
    conversion_rates: {
      BYN: 10,
      USD: 0.1182,
      EUR: 0.09916,
    }
  };

  const server = setupServer(
    rest.get('https://v6.exchangerate-api.com/v6/3402d928822e13d5d341634b/latest/NOK', (req, res, ctx) => {
      return res(ctx.json(response))
    }),
  );

  beforeAll(() => server.listen());
  afterAll(() => server.close());
  afterEach(() => server.resetHandlers());

  let country = {
      "id": 1,
      "currency": "NOK",
    }
  ;

  it('renders correctly', async () => {

    const {container, findByText} = render(<ExchangeWidget country={country} />);

    expect(await findByText("10")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});