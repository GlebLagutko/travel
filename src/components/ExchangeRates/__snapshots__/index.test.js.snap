// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeWidget renders correctly 1`] = `
<div>
  <div
    class="exchange"
  >
    <div
      style="color: black;"
    >
      <strong>
        NOK
      </strong>
    </div>
    <div
      class="exchange-div"
    >
      <div>
        <strong>
          USD
        </strong>
         : 
        <span>
          0.1182
        </span>
      </div>
      <div>
        <strong>
          EUR
        </strong>
         : 
        <span>
          0.09916
        </span>
      </div>
      <div>
        <strong>
          BYN
        </strong>
         : 
        <span>
          10
        </span>
      </div>
    </div>
  </div>
</div>
`;
